--- origsrc/gnome-do-0.8.3.1/Do/Do.exe.config.in	2009-06-28 19:19:29.000000000 -0500
+++ src/gnome-do-0.8.3.1/Do/Do.exe.config.in	2010-01-04 03:31:15.943039900 -0600
@@ -1,4 +1,3 @@
 <configuration>
-	<dllmap dll="libc" target="libc.so.6"/>
-	<dllmap dll="libdo" target="@expanded_libdir@/@PACKAGE@/libdo.so"/>
+	<dllmap dll="libdo" target="@expanded_libdir@/@PACKAGE@/cygdo.dll"/>
 </configuration>
--- origsrc/gnome-do-0.8.3.1/Do/gnome-do.in	2009-12-07 23:21:50.279510600 -0600
+++ src/gnome-do-0.8.3.1/Do/gnome-do.in	2010-01-04 13:33:13.890868400 -0600
@@ -16,10 +16,7 @@ fi
 dbus-send --session --type=method_call --dest='org.gnome.Do' \
 	'/org/gnome/Do/Controller' 'org.gnome.Do.Controller.Summon'
 
-# If Do is not running, run it.
-if pgrep -u "`id -u`" '^gnome-do$' >/dev/null; then
-	mono "$GNOME_DO_EXE" "$@"
-fi
+mono "$GNOME_DO_EXE" "$@"
 while [ "$?" -eq "20" ]; do
 	mono "$GNOME_DO_EXE" "$@"
 done
--- origsrc/gnome-do-0.8.3.1/Do.Interface.Wink/Do.Interface.Wink.dll.config	2009-06-28 19:19:29.000000000 -0500
+++ src/gnome-do-0.8.3.1/Do.Interface.Wink/Do.Interface.Wink.dll.config	2010-01-04 03:31:00.410151500 -0600
@@ -1,4 +1,4 @@
 <configuration>
-        <dllmap dll="X11" target="libX11.so.6"/>
-        <dllmap dll="libgdk-x11" target="libgdk-x11-2.0.so.0"/>
+        <dllmap dll="X11" target="cygX11-6.dll"/>
+        <dllmap dll="libgdk-x11" target="cyggdk-x11-2.0-0.dll"/>
 </configuration>
--- origsrc/gnome-do-0.8.3.1/build.rules.mk	2009-06-28 19:19:29.000000000 -0500
+++ src/gnome-do-0.8.3.1/build.rules.mk	2010-01-04 11:36:31.182336400 -0600
@@ -24,7 +24,11 @@ OUTPUT_FILES = \
 
 moduledir = $(pkglibdir)
 # Install libraries as data; there's no need for them to be excutable
+if ENABLE_DEBUG
 module_DATA = $(foreach file,$(filter %.dll,$(OUTPUT_FILES)),$(file) $(file).mdb) $(foreach file,$(filter %.exe,$(OUTPUT_FILES)),$(file).mdb)
+else
+module_DATA = $(filter %.dll,$(OUTPUT_FILES))
+endif
 # Install executables as scripts
 module_SCRIPTS = $(filter %.exe,$(OUTPUT_FILES))
 
--- origsrc/gnome-do-0.8.3.1/libdo/Makefile.am	2009-06-28 19:19:29.000000000 -0500
+++ src/gnome-do-0.8.3.1/libdo/Makefile.am	2010-01-04 03:30:00.499724800 -0600
@@ -17,7 +17,7 @@ libdo_la_SOURCES =		\
 	eggaccelerators.h	\
 	eggaccelerators.c
 
-libdo_la_LDFLAGS = -export-dynamic -module -avoid-version
+libdo_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
 libdo_la_LIBADD = $(LIBDO_LIBS)
 
 maintainer-clean-local:
